XenAdmin/SettingsPanels/VBDEditPage.cs

```diff
@@ -370,5 +363,38 @@ public String SubText
                     modeComboBox.SelectedItem);
             }
         }
+
+        public class DevicePositionComboBoxItem
+        {
+            private VBD vbd;
+            private string position;
+
+            public VBD VBD { get { return vbd; } }
+            public string Position { get { return position; } }
+
+            public DevicePositionComboBoxItem(string position, VBD vbd = null)
+            {
+                this.vbd = vbd;
+                this.position = position;
+            }
+
+            public override string ToString()
+            {
+                VDI vdi = null;
+
+                if (vbd != null && vbd.Connection != null)
+                {
+                    vdi = vbd.Connection.Resolve(vbd.VDI);
+                }
+
+                if (vbd == null)
+                    return position;
+
+                if (vdi != null)
+                    return string.Format(Messages.VBD_EDIT_CURRENTLY_IN_USE_BY, position, vdi.ToString());

```

Comment:
The code at line 94-97 was not touched by this PR, I leave code as is, because removing those lines you mentioned would cause a change in functionality, because the "vbd" is a field in declared the class, not just local in the method. (Therefore calling this setter with a non-VBD would null out the previously set one after this change whereas it is not the case now.)